ftlib/consensus/gossip/memberlist/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+    /*
+       #include <stdlib.h>
+       typedef struct member_list{
+           char *addrs[1024];
+           int size;
+       } member_list;
+    */
+    "C"
+    "fmt"
+    "github.com/hashicorp/memberlist"
+    "os"
+)
+
+var list *memberlist.Memberlist
+
+
+func main() {
+}
+
+//export init_memberlist
+func init_memberlist(cLogFileName *C.char) C.int {
+    var err error
+    config := memberlist.DefaultLocalConfig()
+
+    logFileName := C.GoString(cLogFileName)
+
+    if logFileName != "" {
+        f, err := os.Create(logFileName)
+        if err != nil {
+            panic("Error! cannot create log file")
+        }
+        config.LogOutput = f
+    }
+    list, err = memberlist.Create(config)
+    if err != nil {
+        panic("Failed to create memberlist: " + err.Error())
+        return 1
+    }
+    return 0
+}
+
+//export join
+func join(ns []*C.char) C.int {
```

Comment:
What's `ns`? Is this the IP of a known member?

Reply:
yes, a list of ip of known nodes/workers